src/initialize_dae.jl

```diff
@@ -152,20 +146,32 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
 
         nlequation! = @closure (out, u, p) -> begin
             update_coefficients!(M, u, p, t)
-            #M * (u-u0)/dt - f(u,p,t)
+            # f(u,p,t) + M * (u0 - u)/dt
             tmp = isad ? PreallocationTools.get_tmp(_tmp, u) : _tmp
-            @. tmp = (u - u0) / dt
+            tmp .= (u0 .- u) .* inv(dt)
             mul!(_vec(out), M, _vec(tmp))
             f(tmp, u, p, t)
-            out .-= tmp
+            out .+= tmp
             nothing
         end
 
-        nlsolve = default_nlsolve(alg.nlsolve, isinplace, u0, isad)
+        jac = if isnothing(f.jac)
+            f.jac
+        else
+            @closure (J, u, p) -> begin
+                # f(u,p,t) + M * (u0 - u)/dt
+                # df(u,p,t)/du - M/dt
+                f.jac(J, u, p, t)
```

Comment:
for a DAE, `f.jac` takes a gamma argument.
I suspect this should be:
```suggestion
                f.jac(J, u, p, 0.0, t)
```